use/singularity.md

```diff
@@ -78,31 +80,29 @@ The following example illustrates how this may be done with a SLURM job script.
 
 # Create temporary directory to be populated with directories to bind-mount in the container
 # where writable file systems are necessary. Adjust path as appropriate for your computing environment.
-workdir=$(python -c 'import tempfile; print(tempfile.mkdtemp())')
-
-mkdir -p -m 700 ${workdir}/run ${workdir}/tmp ${workdir}/var/lib/rstudio-server
-cat > ${workdir}/database.conf <<END
-provider=sqlite
-directory=/var/lib/rstudio-server
-END
+workdir=$(mktemp -d)
 
 # Set OMP_NUM_THREADS to prevent OpenBLAS (and any other OpenMP-enhanced
 # libraries used by R) from spawning more threads than the number of processors
 # allocated to the job.
 #
-# Set R_LIBS_USER to a path specific to rocker/rstudio to avoid conflicts with
+# Set R_LIBS_USER to an existing path specific to rocker/rstudio to avoid conflicts with
 # personal libraries from any R installation in the host environment
 
 cat > ${workdir}/rsession.sh <<END
 #!/bin/sh
-export OMP_NUM_THREADS=${SLURM_JOB_CPUS_PER_NODE}
-export R_LIBS_USER=${HOME}/R/rocker-rstudio/4.2
+export OMP_NUM_THREADS=${SLURM_CPUS_ON_NODE}
+export R_LIBS_USER=${HOME}/R/rocker-rstudio/4.4.2
+mkdir -p "\${R_LIBS_USER}"
```

Comment:
Maybe typo?

```suggestion
mkdir -p "${R_LIBS_USER}"
```

Reply:
The `\` was used so the resulting rsession.sh script that is generated (which is executed in the container) contains (literally):
```
mkdir -p "${R_LIBS_USER}"
```
instead of expanding R_LIBS_USER in the here-document to the value it has in the host environment. This is admittedly somewhat subtle---which isn't great for readability/maintainability---though there is another instance where a backslash escape is used in the here-document (`exec /usr/lib/rstudio-server/bin/rsession "\${@}"`).

For more consistency, I guess there's no reason why there couldn't be a backslash escape in:
```
export R_LIBS_USER=\${HOME}/R/rocker-rstudio/4.4.2
```
as home directories are `$HOME` on the host should be the same as `$HOME` in the container (the user's home directory is mounted by default in a Singularity container).

I'm probably splitting hairs at this point :-)

Reply:
> The `\` was used so the resulting rsession.sh script that is generated (which is executed in the container) contains (literally):

In that case, can't you use `"END"` instead of `END`?

Reply:
The value of `${SLURM_CPUS_ON_NODE}` from the host was being used in the here-doc. However... it seems this should no longer be necessary for OpenBLAS, which should detect the number of CPUs in the cgroup when OMP_NUM_THREADS is unset (https://github.com/OpenMathLib/OpenBLAS/issues/1155) --- I suspect the vast majority of Slurm clusters today use cgroups to constrain processes to resources allocated in the job script --- and (for other OpenMP applications) libgomp seems to use the number of CPUs in the cgroup for the default number of threads in a parallel region (tested with a small C program compiled/run in both rocker/rstudio:4.0.0 and rocker/rstudio:4.4.2 environments).

I'm going to go out on a limb and remove the explicit OMP_NUM_THREADS environment variable assignment in rsession.sh (and use the cleaner `<<"END"` here-document syntax you suggested).